packages/tracing-processing/src/services/enricherService.ts

```diff
@@ -70,36 +69,33 @@ export function dbServiceBuilder(db: DB) {
               );
 
               if (!consumer) {
-                return {
-                  purposeId: record.purpose_id,
-                  status: record.status,
-                  errorCode: PurposeErrorCodes.CONSUMER_NOT_FOUND,
-                  message: PurposeErrorCodes.CONSUMER_NOT_FOUND,
-                  rowNumber: record.rowNumber,
-                };
+                throw getEnrichedPurposeError(
+                  `Consumer ${tracing.tenantId} not found for tracingId: ${tracing.tracingId}, purpose_id: ${record.purpose_id}`,
+                );
               }
 
               if (!eService) {
-                return {
-                  purposeId: record.purpose_id,
-                  status: record.status,
-                  errorCode: PurposeErrorCodes.ESERVICE_NOT_FOUND,
-                  message: PurposeErrorCodes.ESERVICE_NOT_FOUND,
-                  rowNumber: record.rowNumber,
-                };
+                throw getEnrichedPurposeError(
+                  `Eservice ${fullPurpose.eservice_id} not found for tracingId: ${tracing.tracingId}, purpose_id: ${record.purpose_id}`,
+                );
               }
 
               const tenantEservice = await db.oneOrNone<EserviceSchema>(
-                `SELECT * FROM tracing.eservices WHERE (producer_id = $1 OR consumer_id = $1)`,
+                `SELECT * FROM tracing.eservices WHERE producer_id = $1`,
                 [tracing.tenantId],
               );
+              const tenantPurpose = await db.oneOrNone<EserviceSchema>(
+                `SELECT * FROM tracing.purposes WHERE id = $1 AND consumer_id = $2`,
```

Comment:
do we need another query? we can return the consumer id in the `fullPurpose` and use that